### PR TITLE
Added a new class PaginatedEmbeds

### PR DESCRIPTION
**Pull request by Kshitiz-Arya**
It takes a dictionary containing name and values as key value pair and split them into several embeds according to size provided by the user.

### CODE REVIEW

**Comment by chrisdewa** [comment]
Thanks for the PR, I'll review it soon!

**Comment by Kshitiz-Arya** [comment]
I have not tested this for edge cases, so it is ok if you can test them first.

File: dpytools/embeds.py [hunk 1]
```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
         """
         for name, value in kwargs.items():
             self.add_field(name=name, value=value, inline=inline)
+
+            
+class EmbedList:
+    
+    """A class that takes dictionary containing name and value and splint them into several Embeds.
```
Comment (chrisdewa) [inline dpytools/embeds.py:126]: If this class only really focuses on paginating fields, perhaps its better to change its name and description to that.

File: dpytools/embeds.py [hunk 2]
```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
...
+            raise ValueError("Number of Embed fields can't be zero")
+
+        for i in range(0, len(tuple_list), num):
+            yield tuple_list[i:i+num]
+    
+    def _add_embed(self, dicts):
+        """Add embeds to pages
+
+        Args:
+            dicts :- Name, Value dictionary to generate embed from
```
Comment (chrisdewa) [inline dpytools/embeds.py:232]: It's improtant to specify that name and value correspond to each field's key and value.

File: dpytools/embeds.py [hunk 3]
```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
...
+
+        Returns:
+            None
+
+        """        
+        for d in self._chunks(dicts):
```
Comment (chrisdewa) [inline dpytools/embeds.py:238]: Same as before, variable names in this loop are too ambiguous. Try to name them something that actually tells what they are or what they do

File: dpytools/embeds.py [hunk 4]
```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
         """
         for name, value in kwargs.items():
             self.add_field(name=name, value=value, inline=inline)
+
+            
+class EmbedList:
```
Comment (chrisdewa) [inline dpytools/embeds.py:124]: Overall i like where this class is heading but i thing it's missing some important things. For example, author url and icon_url, or the embed url, etc. The class should be able to make a full Embed. 

A suggestion could be accepting a `base_embed` argument on `__init__` that contains everyting but the fields and just paginate the fields using the base embed as template.

**Review by chrisdewa — CHANGES_REQUESTED** [review]
I like where his class is heading but its missing some important features. 
Also, i think what this class provide is an easy way to paginate multiple fields into whatever amount of embeds needed. Perpahs it'd be better if the class used what the base Embed subclass, provided in this library, can do and expanded it with the pagination.

Please address them for futher review.

**Comment by Kshitiz-Arya** [comment]
I have made few changes to to implementation of EmbedList, which has been renamed to PaginatedEmbeds (I am bad at naming things so feel free to make any changes here)

File: dpytools/embeds.py [hunk 5]
```diff
@@ -121,50 +121,40 @@ def add_fields(self, inline=True, **kwargs):
...
-    def __init__(self, **options):
-        self.dicts = options.pop('dicts')
-        self.title = options.pop('title', '')
-        self.description = options.pop('description', '')
-        self.footer = options.pop('footer', '')
-        self.author = options.pop('author', '')
-        self.inline = options.pop('inline', True)
-        self.colour = options.pop('colour', 0)
-        self.size = options.pop('size', 25)
+    def __init__(self, embed:Embed, fields_dict:Dict[str, str],size:int=25, inline:bool=True):
```
Comment (Kshitiz-Arya) [inline dpytools/embeds.py:141]: Now taking an user-initialized embed to act as a template for paginated embed

File: dpytools/embeds.py [hunk 6]
```diff
@@ -118,3 +119,141 @@ def add_fields(self, inline=True, **kwargs):
...
+        self.description = options.pop('description', '')
+        self.footer = options.pop('footer', '')
+        self.author = options.pop('author', '')
+        self.inline = options.pop('inline', True)
+        self.colour = options.pop('colour', 0)
+        self.size = options.pop('size', 25)
+        
+        self.field_limit = 25
+        self.char_limit = 6000
+        self.clear()
```
[the remark this answers is not in the document]
Comment (Kshitiz-Arya) [reply dpytools/embeds.py]: Made this change

**Comment by chrisdewa** [comment]
Thanks for the corrections! I review this today, i hope, i just ask that you pull the lastest changes from master so there's no conflicts with the PR!

**Comment by chrisdewa** [comment]
Finally merging!
